lib/msprime.c

```diff
@@ -5863,177 +5824,106 @@ msp_dirac_common_ancestor_event(msp_t *self, population_id_t pop_id, label_id_t
 static double
 beta_model_time_to_generations(simulation_model_t *model, double t)
 {
-    return 4 * model->params.beta_coalescent.scalar * t;
+    double alpha = model->params.beta_coalescent.alpha;
+    double truncation_point = model->params.beta_coalescent.truncation_point;
+    double reference_size = model->reference_size;
+    double m = 2.0 + exp(alpha * log(2) + (1 - alpha) * log(3) - log(alpha - 1));
+    double timescale = exp(log(alpha) - alpha * log(m)
+        - (alpha - 1) * log(reference_size))
+        * gsl_sf_beta_inc(2 - alpha, alpha, truncation_point);
```

Comment:
There's a fair bit of repeated code here. How about 
```c
static double 
beta_model_compute_timescale(simulation_model_t *model)
{   
    double alpha = model->params.beta_coalescent.alpha;
￼    double truncation_point = model->params.beta_coalescent.truncation_point;
￼    double reference_size = model->reference_size;
￼    double m = 2.0 + exp(alpha * log(2) + (1 - alpha) * log(3) - log(alpha - 1));
￼    double timescale = exp(log(alpha) - alpha * log(m)
￼        - (alpha - 1) * log(reference_size))
￼        * gsl_sf_beta_inc(2 - alpha, alpha, truncation_point);
    return timescale;
}

static double
￼beta_model_time_to_generations(simulation_model_t *model, double t)
￼{
    return  4 * t * beta_model_compute_timescale(model);
}
```

Reply:
Good idea. Added.